### PR TITLE
add score and display

### PR DESCRIPTION
**Pull request by freddykaberuka**
## Completed Task

- use Api endpoints
- add refresh button functionality
- add score through form

### CODE REVIEW

File: src/score.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+getScore = () => {
+  const listScore = document.querySelector('.list');
+  listScore.innerHTML = this.dataScore.map((el, index) => `<div class=${index % 2 !== 0 ? 'list1' : 'list2'}>${el.user}:${el.score}</div>`).join('');
+}
+
+fetchScore = async () => {
+  try {
+    const data = await fetch(this.baseUrl);
+    const response = await data.json();
+    response.result.map((item) => this.dataScore.push(item));
```
Comment (ShahierNashaat) [inline src/score.js]: - [x] Please make sure to reset the dataScore to an empty array as when you push inside it you are adding the data again inside it or you can optimize it without using the map function you can use the spread operator example: `dataScore = [...response.result];`.

File: src/score.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user, score }),
+    };
+    const data = await fetch(this.baseUrl, config);
+    const response = await data.json();
+    this.dataScore.push(response);
+    return this.fetchScore();
```
Comment (ShahierNashaat) [inline src/score.js]: - [x] Kindly be noticed that you should add the new score inside the dataScore in the addScore function or call the fetchScore function as you case the score to be added two times in the dataScore array.

File: src/index.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
-// eslint-disable-next-line no-unused-vars
-import _ from 'lodash';
+import Score from './score.js';
 import './style.css';
 
-const component = () => {
+const newScore = new Score();
```
Comment (ShahierNashaat) [inline src/index.js:4]: - [x] Kindly when the program starts call the newScore.fetchScore to view the data to the user in the beginning.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @freddykaberuka , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (freddykaberuka) [reply src/index.js:4]: Hi @ShahierNashaat ,
does this require to work when you click on refresh button?

Comment (ShahierNashaat) [reply src/index.js:4]: No you just call the function when the website loads and when click the refresh button

**Review by Meltrust — APPROVED** [review]
Hi @freddykaberuka,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for refactoring your addScore function, and for showing the updated scores when the app starts.✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**